flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/JdbcOutputFormat.java

```diff
@@ -154,7 +154,9 @@ public void open(int taskNumber, int numTasks) throws IOException {
                                         try {
                                             flush();
                                         } catch (Exception e) {
-                                            flushException = e;
+                                            if (flushException == null) {
```

Comment:
What do you think about:
`flushException = ExceptionUtils.firstOrSuppressed(e, flushException);`

Reply:
This does not solve the chaining exception problem. We'll still get a lot of suppressed `RuntimeException`s.

If we need to store all exceptions and avoid this problem we'll need a special exception class so that when we catch this exception class in timer we just ignore it.

Reply:
I think here is the situation that we only need to store the original exception. Since the following exceptions are all  produced by `checkFlushException()` with no further information. 
So, I think it is ok to just ignore the following exceptions without suppress them.

Reply:
OK, Thanks for the explanation. I realized what you are talking about. ^_^

Reply:
@tsreaper I agree with you.
You are discussing about whether multiple exceptions should be stored to show. We can consider two scenerios.
First, the exceptions are triggered by the same one row. In this case, the exceptions are almost the same one (except the extreme case like a temporary connection exception followed by a sql exception from the database, considering it occurs rarely, I think it's ok to just store the first one to trigger a job restart, and after restart, user will see the next exception if it still there), so only one exception need to be stored.
Second, the exceptions are triggered by different rows. Consider a case that row1 causes exception A followed by row2 causes exception B. In such case, if exception B do not occur, row2 will be processed and emit to the downstream. So I think if row1 triggers an exception, no rows after row1 should be processed and therefore no further exceptions should be triggered.
So, I think we do not need to catch multiple exceptions to show to the user.

Reply:
@JingsongLi Could you take a look at this, if you have some time. Thanks!